File: 0136-single-number/0136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let set = new Set()
    for(const num of nums){
        if(!set.has(num)) set.add(num)
        else set.delete(num)
    }
    
    return set.values().next().value
};

File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        num_set = set()
        for num in nums:
            if not num in num_set:
                num_set.add(num)
            else:
                num_set.remove(num)
        return list(num_set)[0]